interop-java/src/main/java/scalaz/zio/java/data/Either.java

```diff
@@ -0,0 +1,47 @@
+package scalaz.zio.java.data;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public final class Either<L,R> {
+    public static <L,R> Either<L,R> left(L value) {
+        return new Either<>(Optional.of(value), Optional.empty());
+    }
+
+    public static <L,R> Either<L,R> right(R value) {
+        return new Either<>(Optional.empty(), Optional.of(value));
+    }
+
+    public static <L, R> Either<L, R> fromScala(scala.util.Either<L, R> either) {
+        return either.fold(Either::left, Either::right);
+    }
+
+    private final Optional<L> left;
+    private final Optional<R> right;
```

Comment:
This is not a safe representation, because what if both sides have something, or neither side does? It's better to create anonymous classes for `left` and `right` that implement the `fold` method. Everything can be done atop `fold`.